src/services/evm/erc20_handler.ts

```diff
@@ -137,6 +139,57 @@ export class Erc20Handler {
     }
   }
 
+  static buildTransferActivityByCosmos(e: Event) {
+    try {
+      const getAddressFromAttrAndConvert0x = (
+        attrs: EventAttribute[],
+        key: string
+      ) => {
+        const attr = attrs.find((attr) => attr.key === key);
+        if (attr) {
+          let { value } = attr;
+          if (value.startsWith('0x')) value = value.toLowerCase();
+          else
+            value = convertBech32AddressToEthAddress(
+              config.networkPrefixAddress,
+              value
+            ).toLowerCase();
+          return value;
+        }
+        return undefined;
+      };
+      const from = getAddressFromAttrAndConvert0x(
+        e.attributes,
+        EventAttribute.ATTRIBUTE_KEY.SENDER
+      );
+      const to = getAddressFromAttrAndConvert0x(
+        e.attributes,
+        EventAttribute.ATTRIBUTE_KEY.RECEIVER
+      );
+      const amount = e.attributes.find(
+        (attr) => attr.key === EventAttribute.ATTRIBUTE_KEY.AMOUNT
+      );
+      const address = getAddressFromAttrAndConvert0x(
+        e.attributes,
+        EventAttribute.ATTRIBUTE_KEY.ERC20_TOKEN
+      );
+      return Erc20Activity.fromJson({
```

Comment:
From/to anh đang lấy sai giá trị:
- Nếu là convert_coin thì from là address của erc20 module account, to là receiver
- Nếu là convert_erc20 thì from là sender và to là erc20 module account

Reply:
nếu thế thì sẽ miss mất activity từ module account sang account cần chuyển đến đúng không, nên chỗ này anh không chỉnh gì, lấy y nguyên từ attribute của nó về thôi
https://explorer.dev.aura.network/tx/C804E36636AC875B6B578469473E9EB57BF1F580FAB8E0C2488CFF1565740720
```
{
          "type": "convert_coin",
          "attributes": [
            {
                "key": "sender",
                "index": true,
                "value": "aura1avtv54auz3n95jcgh8d46z6hzymu6lgamuxpae"
            },
            {
                "key": "receiver",
                "index": true,
                "value": "0xd0ec4cd02018245FFb0dB28E99C4e459C3cD32f0"
            },
            {
                "key": "amount",
                "index": true,
                "value": "1000000"
            },
            {
                "key": "cosmos_coin",
                "index": true,
                "value": "ibc/40CA5EF447F368B7F2276A689383BE3C427B15395D4BF6639B605D36C0846A20"
            },
            {
                "key": "erc20_token",
                "index": true,
                "value": "0xd567B3d7B8FE3C79a1AD8dA978812cfC4Fa05e75"
            }
         ]
      }
```

Reply:
Convert coin  có 2 phase rõ ràng:
- Phase 1: Sender chuyển native coin vào account module
- Phase 2: Module account mint erc20 chuyển đến receiver 
Nếu anh tạo ra activity như hiện tại thì erc20 balance (job handler tính toán phía sau) của sender sẽ bị trừ và receiver sẽ cộng thêm => sai vì bản chất sender mất native coin

Reply:
ừ cũng có lý, như thế thì 1 tx convert coin/erc20 đúng thì sẽ tạo ra 2 activity mới đúng

Reply:
anh sẽ lưu cái from/to là erc20 module account nhé @phamphong9981 , cơ mà lưu là địa chỉ aura thì mấy job sau của em có chạy oke không?

Reply:
anh lưu địa chỉ erc20 module là địa chỉ 0x luôn nhé